File: front-end/ivan-ui/src/types/event.ts
// Backend-aligned Event DTOs for Organization Event Management

export interface EventDto {
  eventId: number;
  organizationId: number;
  organizationName: string;
  eventName: string;
  categoryId: number;
  categoryName: string;
  statusId: number;
  statusName: string;
  description: string;
  shortDescription?: string;
  startDate: string;
  endDate: string;
  registrationStartDate?: string;
  registrationEndDate?: string;
  location: string;
  detailedAddress?: string;
  wardCommune?: string;
  district?: string;
  province?: string;
  maxVolunteers: number;
  minVolunteers?: number;
  currentVolunteers?: number;
  requiredSkills?: string;
  ageRequirement?: string;
  genderRequirement?: string;
  requirements?: string;
  benefits?: string;
  contactPerson?: string;
  contactPhone?: string;
  contactEmail?: string;
  bannerImageUrl?: string;
  galleryImages?: string;
  isFeatured?: boolean;
  isUrgent?: boolean;
  priority?: number;
  budget?: number;
  currency?: string;
  viewCount?: number;
  registrationCount?: number;
  rating?: number;
  ratingCount?: number;
  createdAt?: string;
  updatedAt?: string;
  createdBy?: string;
  updatedBy?: string;
  isActive?: boolean;
}

export interface CreateEventDto {
  organizationId?: number; // Auto-set by backend
  eventName: string;
  categoryId: number;
  statusId?: number;
  description: string;
  shortDescription?: string;
  startDate: string;
  endDate: string;
  registrationStartDate?: string;
  registrationEndDate?: string;
  location: string;
  detailedAddress?: string;
  wardCommune?: string;
  district?: string;
  province?: string;
  maxVolunteers: number;
  minVolunteers?: number;
  requiredSkills?: string;
  ageRequirement?: string;
  genderRequirement?: string;
  requirements?: string;
  benefits?: string;
  contactPerson?: string;
  contactPhone?: string;
  contactEmail?: string;
  bannerImageUrl?: string;
  galleryImages?: string;
  isFeatured?: boolean;
  isUrgent?: boolean;
  priority?: number;
  budget?: number;
  currency?: string;
}

export interface UpdateEventDto extends Partial<CreateEventDto> {
  eventId: number;
}

export interface EventFilterDto {
  search?: string;
  categoryIds?: number[];
  statusIds?: number[];
  startDateFrom?: string;
  startDateTo?: string;
  endDateFrom?: string;
  endDateTo?: string;
  province?: string;
  district?: string;
  isFeatured?: boolean;
  isUrgent?: boolean;
  minVolunteers?: number;
  maxVolunteers?: number;
  page: number;
  size: number;
  sortBy: string;
  sortDirection: string;
}

export interface EventStatsDto {
  totalEvents: number;
  planningEvents: number;
  activeEvents: number;
  inProgressEvents: number;
  completedEvents: number;
  cancelledEvents: number;
  totalVolunteers: number;
  totalRegistrations: number;
  averageRating: number;
  upcomingEventsThisMonth: number;
  registrationRate: number;
  eventsByCategory: Record<string, number>;
  eventsByMonth: Record<string, number>;
}

export interface EventCategoryDto {
  categoryId: number;
  categoryName: string;
  description?: string;
  isActive: boolean;
}

export interface EventStatusDto {
  statusId: number;
  statusName: string;
  description?: string;
  isActive: boolean;
}

export interface UpdateEventStatusDto {
  statusId: number;
  reason?: string;
}

// Event Analytics & Transitions
export interface EventAnalyticsDto {
  eventId: number;
  timeframe: string;
  registrations: {
    total: number;
    approved: number;
    pending: number;
    rejected: number;
    trend: number;
  };
  volunteers: {
    current: number;
    target: number;
    fulfillmentRate: number;
  };
  engagement: {
    viewCount: number;
    rating: number;
    reviews: number;
  };
  timeline: Array<{
    date: string;
    registrations: number;
    views: number;
  }>;
}

File: back-end/ivan-api/ivan-api/Models/Event.cs
using System;
using System.Collections.Generic;

namespace ivan_api.Models;

public partial class Event
{
    public int EventId { get; set; }

    public int OrganizationId { get; set; }

    public string EventName { get; set; } = null!;

    public int CategoryId { get; set; }

    public int StatusId { get; set; }

    public string? Description { get; set; }

    public string? ShortDescription { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    public DateTime? RegistrationStartDate { get; set; }

    public DateTime? RegistrationEndDate { get; set; }

    public string? Location { get; set; }

    public string? DetailedAddress { get; set; }

    public string? WardCommune { get; set; }

    public string? District { get; set; }

    public string? Province { get; set; }

    public decimal? Latitude { get; set; }

    public decimal? Longitude { get; set; }

    public int? MaxVolunteers { get; set; }

    public int? MinVolunteers { get; set; }

    public int? CurrentVolunteers { get; set; }

    public string? RequiredSkills { get; set; }

    public string? AgeRequirement { get; set; }

    public string? GenderRequirement { get; set; }

    public string? Requirements { get; set; }

    public string? Benefits { get; set; }

    public string? ContactPerson { get; set; }

    public string? ContactPhone { get; set; }

    public string? ContactEmail { get; set; }

    public string? BannerImageUrl { get; set; }

    public string? GalleryImages { get; set; }

    public bool? IsFeatured { get; set; }

    public bool? IsUrgent { get; set; }

    public int? Priority { get; set; }

    public int? ViewCount { get; set; }

    public int? RegistrationCount { get; set; }

    public int? CompletedVolunteers { get; set; }

    public decimal? Rating { get; set; }

    public int? RatingCount { get; set; }

    public decimal? Budget { get; set; }

    public string? Currency { get; set; }

    public bool? IsActive { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public int? CreatedBy { get; set; }

    public int? UpdatedBy { get; set; }

    // Additional properties for AI features
    public string? EventType { get; set; }
    
    public int? MaxParticipants { get; set; }
    
    public int? CreatedByUserId { get; set; }

    public virtual EventCategory Category { get; set; } = null!;

    public virtual ICollection<Certificate> Certificates { get; set; } = new List<Certificate>();

    public virtual ICollection<CoordinatorSchedule> CoordinatorSchedules { get; set; } = new List<CoordinatorSchedule>();

    public virtual ICollection<CoordinatorTask> CoordinatorTasks { get; set; } = new List<CoordinatorTask>();

    public virtual User? CreatedByNavigation { get; set; }

    public virtual ICollection<EventRegistration> EventRegistrations { get; set; } = new List<EventRegistration>();

    public virtual ICollection<Feedback> Feedbacks { get; set; } = new List<Feedback>();

    public virtual ICollection<OnSiteTask> OnSiteTasks { get; set; } = new List<OnSiteTask>();

    public virtual Organization Organization { get; set; } = null!;

    public virtual EventStatus Status { get; set; } = null!;

    public virtual User? UpdatedByNavigation { get; set; }

    public virtual ICollection<VolunteerSchedule> VolunteerSchedules { get; set; } = new List<VolunteerSchedule>();
}
